### PR TITLE
refactor: make StoptimeOutOfOrderTimesValidator extend SingleFileValidator

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR provides changes to extract the rule that checks the order of times in `stop_times.txt` as a new rule that extends `SingleFileValidator`  class. 

**Expected behavior:** 

Unit tests have been adapted to this refactor, but no logic has changed. Therefore the same logic for existing tests should pass. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StoptimeOutOfOrderTimesValidator.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *   <li>{@link StopTimeWithDepartureBeforeArrivalTimeNotice} - departure_time &lt; arrival_time
+ * </ul>
+ */
+@GtfsValidator
+public class StoptimeOutOfOrderTimesValidator extends SingleEntityValidator<GtfsStopTime> {
+
+  @Override
+  public void validate(GtfsStopTime stopTime, NoticeContainer noticeContainer) {
+    if (stopTime.hasDepartureTime() && stopTime.hasArrivalTime()) {
+      if (stopTime.departureTime().isBefore(stopTime.arrivalTime())) {
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StoptimeOutOfOrderTimesValidator.java:39]: Please collapse that to a single "if" here.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StoptimeOutOfOrderTimesValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    underTest.validate(
+        createStopTime(
+            0,
+            GtfsTime.fromSecondsSinceMidnight(340),
+            GtfsTime.fromSecondsSinceMidnight(518)
+        ),
+        noticeContainer);
+    assertThat(noticeContainer.getValidationNotices()).isEmpty();
+  }
```
Comment (aababilov) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StoptimeOutOfOrderTimesValidatorTest.java:40]: Please make it more concise like here:

```
private static List<ValidationNotice> generateNotices(GtfsAttribution attribution) {
    NoticeContainer noticeContainer = new NoticeContainer();
    AttributionWithoutRoleValidator validator = new AttributionWithoutRoleValidator();
    validator.validate(attribution, noticeContainer);
    return noticeContainer.getValidationNotices();
  }

  @Test
  public void attributionWithoutRoleShouldGenerateNotice() {
    assertThat(
            generateNotices(
                new GtfsAttribution.Builder()
                    .setCsvRowNumber(2)
                    .setAttributionId("attr-1")
                    .build()))
        .containsExactly(new AttributionWithoutRoleNotice(2, "attr-1"));
  }
```

**Comment by aababilov** [comment]
Make sense, Lionel!

What came to my mind: that may be a basic validation in Core. I will add a new annotation to validate that times and dates are in order.

**Review by aababilov — CHANGES_REQUESTED** [review]

**Comment by aababilov** [comment]
Here is an annotation-based solution: https://github.com/MobilityData/gtfs-validator/pull/813

**Comment by lionel-nj** [comment]
Closing as implemented in #813.

**Pull request closed without merging**